python/bind_data_access.h

```diff
@@ -120,6 +120,15 @@ class DataAccessHelper {
   };
 };
 
+inline void expect_scalar(const Dimensions &dims, const std::string_view name) {
+  if (dims != Dimensions{}) {
+    std::ostringstream oss;
+    oss << "The '" << name << "' property cannot be used with non-scalar "
+        << "Variables. Got dimensions " << to_string(dims);
+    throw std::runtime_error(oss.str());
```

Comment:
Is this a different exception type now?

Reply:
Yes. It used to be a `MismatchError`. But AFAIK this formats its own, kind of confusing, error message.

Reply:
Throw `except::DimensionError` then.

Reply:
Done, but it still translates into `RuntimeError` in Python.